internal/cleanup/cleanup.go

```diff
@@ -91,6 +91,20 @@ func (h *exposureCleanupHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 
 	stats.Record(ctx, cleanup.ExposuresDeleted.M(count))
 	logger.Infof("cleanup run complete, deleted %v records.", count)
+
+	// Clean up associated stats.
+	statsCount, err := h.database.DeleteStatsBefore(timeoutCtx, cutoff)
```

Comment:
We may need to increase the allowed latency on this job now.

Reply:
right now, the job timeout is 5 minutes - I'm not worried this will increase that.
current runtimes are about ~10s. We could increase the frequency of the exposure cleanup as well.